src/vs/editor/contrib/hover/browser/hoverTypes.ts

```diff
@@ -103,6 +103,10 @@ export interface IEditorHoverRenderContext {
 	 * The status bar for actions for this hover.
 	 */
 	readonly statusBar: IEditorHoverStatusBar;
+	/**
+	 * The disposable store for this hover
+	 */
+	readonly disposables: DisposableStore | undefined;
```

Comment:
This is unclear to me.
Is the `IEditorHoverRenderContext` disposables? How calls `disposables.dispose()`?

Reply:
Yes so the disposable store is created here:

https://github.com/microsoft/vscode/blob/dd42b9fda56ee3c2f37cf94941a1220c213aca4f/src/vs/editor/contrib/hover/browser/contentHover.ts#L207

Then it is passed in the `showAt` method here in the hoverData parameter.

https://github.com/microsoft/vscode/blob/dd42b9fda56ee3c2f37cf94941a1220c213aca4f/src/vs/editor/contrib/hover/browser/contentHover.ts#L246

The hoverData is saved in the field `this._hoverData`. When the hover is hidden, we call `_setHoverData` with undefined:

https://github.com/microsoft/vscode/blob/dd42b9fda56ee3c2f37cf94941a1220c213aca4f/src/vs/editor/contrib/hover/browser/contentHover.ts#L846

Finally inside of `_setHoverData` we dispose the store:

https://github.com/microsoft/vscode/blob/dd42b9fda56ee3c2f37cf94941a1220c213aca4f/src/vs/editor/contrib/hover/browser/contentHover.ts#L756